patches/minecraft/net/minecraft/world/gen/feature/jigsaw/JigsawManager.java.patch

```diff
@@ -0,0 +1,63 @@
+--- a/net/minecraft/world/gen/feature/jigsaw/JigsawManager.java
++++ b/net/minecraft/world/gen/feature/jigsaw/JigsawManager.java
+@@ -39,6 +39,10 @@
+    }
+ 
+    public static void func_236823_a_(ResourceLocation p_236823_0_, int p_236823_1_, JigsawManager.IPieceFactory p_236823_2_, ChunkGenerator p_236823_3_, TemplateManager p_236823_4_, BlockPos p_236823_5_, List<? super AbstractVillagePiece> p_236823_6_, Random p_236823_7_, boolean p_236823_8_, boolean p_236823_9_) {
++      func_236823_a_(p_236823_0_, p_236823_1_, p_236823_2_, com.google.common.collect.Maps.newHashMap(), p_236823_3_, p_236823_4_, p_236823_5_, p_236823_6_, p_236823_7_,  p_236823_8_,  p_236823_9_);
```

Comment:
is an overloaded method necessary? Why not create an empty map on Assembler constructor

Reply:
For modded structures, you can now call `func_236823_a_` and provide existing settings.

Reply:
yes that seems ok. So that you can have preset entries into the assembersettings e.g. based on the feature config (exact 2 diamondhouses or 3)

Reply:
Yes, that is also one of use-cases.